cmd/argocd/commands/app.go

```diff
@@ -1343,6 +1348,17 @@ func NewApplicationDiffCommand(clientOpts *argocdclient.ClientOptions) *cobra.Co
 			argoSettings, err := settingsIf.Get(ctx, &settings.SettingsQuery{})
 			errors.CheckError(err)
 			diffOption := &DifferenceOption{}
+
+			// Server side diff with local requires server side generate to be set as there will be a mismatch with client-generated manifests.
+			if serverSideDiff && local != "" && !serverSideGenerate {
+				log.Fatal("--server-side-diff with --local requires --server-side-generate.")
+			}
+
+			// If the application has ServerSideDiff=true annotation, provide note, but preserve control to the user.
+			if !serverSideDiff && resourceutil.HasAnnotationOption(app, argocommon.AnnotationCompareOptions, "ServerSideDiff=true") {
+				fmt.Printf("Note: Application has ServerSideDiff=true annotation. Consider using --server-side-diff\n\n")
```

Comment:
why are we using `Printf` here? I don't see any format specifiers used either. Let's switch to using `Println` instead.